firestore/integration_test_internal/src/source_test.cc

```diff
@@ -13,511 +13,514 @@ namespace firestore {
 
-  Future<DocumentSnapshot> future = docRef.Get(Source::kServer);
+  Future<DocumentSnapshot> future = doc_ref.Get(Source::kServer);
   Await(future);
   EXPECT_EQ(future.status(), FutureStatus::kFutureStatusComplete);
   EXPECT_EQ(future.error(), Error::kErrorUnavailable);
 }
 
-TEST_F(SourceTest, TestGetCollectionWhileOfflineWithSourceEqualToServer) {
-  const std::map<std::string, MapFieldValue> initialDocs = {
+TEST_F(SourceTest, GetCollectionWhileOfflineWithSourceEqualToServer) {
+  std::map<std::string, MapFieldValue> initial_docs = {
       {"doc1", {{"key1", FieldValue::String("value1")}}},
       {"doc2", {{"key2", FieldValue::String("value2")}}},
       {"doc3", {{"key3", FieldValue::String("value3")}}}};
-  CollectionReference colRef = Collection(initialDocs);
+  CollectionReference col_ref = Collection(initial_docs);
 
-  WaitFor(colRef.Get());
-  WaitFor(TestFirestore()->DisableNetwork());
+  Await(col_ref.Get());
+  DisableNetwork();
 
-  Future<QuerySnapshot> future = colRef.Get(Source::kServer);
+  Future<QuerySnapshot> future = col_ref.Get(Source::kServer);
   Await(future);
   EXPECT_EQ(future.status(), FutureStatus::kFutureStatusComplete);
   EXPECT_EQ(future.error(), Error::kErrorUnavailable);
 }
 
-TEST_F(SourceTest, TestGetDocumentWhileOfflineWithDifferentGetOptions) {
-  MapFieldValue initialData = {{"key", FieldValue::String("value")}};
-  DocumentReference docRef = DocumentWithData(initialData);
+TEST_F(SourceTest, GetDocumentWhileOfflineWithDifferentGetOptions) {
+  MapFieldValue initial_data = {{"key", FieldValue::String("value")}};
+  DocumentReference doc_ref = DocumentWithData(initial_data);
 
-  WaitFor(docRef.Get());
-  WaitFor(TestFirestore()->DisableNetwork());
+  Await(doc_ref.Get());
+  DisableNetwork();
 
-  // Create an initial listener for this query (to attempt to disrupt the
+  // Create an initial listener for this query (to attempt to disrupt the gets
   // below) and wait for the listener to deliver its initial snapshot before
   // continuing.
-  std::promise<Error> errorPromise;
-  std::future<Error> errorFuture = errorPromise.get_future();
-
-  docRef.AddSnapshotListener([&errorPromise](const DocumentSnapshot& snapshot,
-                                             Error error_code,
-                                             const std::string& error_message) {
-    errorPromise.set_value(error_code);
-  });
-  errorFuture.wait();
-
-  Future<DocumentSnapshot> future = docRef.Get(Source::kCache);
-  WaitFor(future);
+  std::promise<Error> error_promise;
+  std::future<Error> error_future = error_promise.get_future();
+
+  doc_ref.AddSnapshotListener(
+      [&error_promise](const DocumentSnapshot& snapshot, Error error_code,
+                       const std::string& error_message) {
+        error_promise.set_value(error_code);
+      });
+  // Note that future::get() will wait until the future has a valid result and
+  // retrieves it. Calling wait() before get() is not needed.
```

Comment:
Ultranit: I think it should be `s/retrieves/retrieve/` (the way I read it, it connects to the previous `will`: `will wait... and (will) retrieve it`). Feel free to disagree.